pkg/resource/segment/download.go

```diff
@@ -42,44 +43,77 @@ func NewDownloadAPI(segmentSource DownloadSource) *DownloadAPI {
 }
 
 func (a DownloadAPI) Download(ctx context.Context, projectName string) (project.ConfigsPerType, error) {
-	log.InfoContext(ctx, "Downloading segments")
+	slog.InfoContext(ctx, "Downloading segments")
 	result := project.ConfigsPerType{}
 
+	lg := slog.With(log.TypeAttr(config.SegmentID))
 	downloadedConfigs, err := a.segmentSource.GetAll(ctx)
 	if err != nil {
-		log.With(log.TypeAttr(config.SegmentID), log.ErrorAttr(err)).ErrorContext(ctx, "Failed to fetch the list of existing segments: %v", err)
+		lg.ErrorContext(ctx, "Failed to fetch the list of existing segments", log.ErrorAttr(err))
 		return nil, nil
 	}
 
+	countReadyMade := 0
 	var configs []config.Config
 	for _, downloadedConfig := range downloadedConfigs {
-		c, err := createConfig(projectName, downloadedConfig)
+		jsonConfig, err := unmarshalConfig(downloadedConfig)
 		if err != nil {
-			log.With(log.TypeAttr(config.SegmentID), log.ErrorAttr(err)).ErrorContext(ctx, "Failed to convert segment: %v", err)
+			lg.ErrorContext(ctx, "Failed to convert segment", log.ErrorAttr(err))
```

Comment:
I think we should change the message here to indicate that parsing the JSON failed. Currently, it is the same message as line 70 below.